openmls_rust_crypto/src/provider.rs

```diff
@@ -262,8 +267,14 @@ impl OpenMlsCrypto for RustCrypto {
                 .map_err(|_| CryptoError::InvalidSignature)
             }
             SignatureScheme::ED25519 => {
-                let k = ed25519_dalek::PublicKey::from_bytes(pk)
+                if pk.len() != ed25519_dalek::PUBLIC_KEY_LENGTH {
+                    return Err(CryptoError::CryptoLibraryError);
+                }
+                let mut k = [0u8; ed25519_dalek::PUBLIC_KEY_LENGTH];
+                k.clone_from_slice(pk);
```

Comment:
Why do you need to copy it in `k` since you tested before that `pk` has the right length ?